x-pack/plugins/apm/server/routes/suggestions/route.ts

```diff
@@ -30,16 +39,31 @@ const suggestionsRoute = createApmServerRoute({
     const size = await context.core.uiSettings.client.get<number>(
       maxSuggestions
     );
-    const suggestions = await getSuggestions({
-      field,
-      searchAggregatedTransactions,
-      setup,
-      size,
-      string,
-    });
+
+    const suggestions = serviceName
+      ? await getSuggestionsWithTermsAggregation({
+          fieldName,
+          fieldValue,
+          serviceName,
+          setup,
+          size,
+          start,
+          end,
+        })
+      : await getSuggestions({
+          fieldName,
+          fieldValue,
+          searchAggregatedTransactions,
```

Comment:
What are you unsure about specifically? if it is needed?

Reply:
that's right. I'm not sure If needed here (or when it's needed)

Reply:
I think it's okay to use it here. It will skip the transaction indices if metrics are being used. There's not much of a performance gain but I think it's more _correct_ to ignore transaction data, although the same environments should be available on transaction metrics.